EnoEngine/FlagSubmissionEndpoint.cs

```diff
@@ -25,34 +25,55 @@ namespace EnoEngine.FlagSubmission
 {
     class FlagSubmissionEndpoint
     {
-        private static readonly Dictionary<long, Channel<(Flag Flag, TaskCompletionSource<FlagSubmissionResult> FeedbackSource)>> Channels =
+        private readonly Dictionary<long, Channel<(Flag Flag, TaskCompletionSource<FlagSubmissionResult> FeedbackSource)>> Channels =
             new Dictionary<long, Channel<(Flag, TaskCompletionSource<FlagSubmissionResult>)>>();
+        private readonly Dictionary<long, TeamFlagSubmissionStatistic> SubmissionStatistics = new Dictionary<long, TeamFlagSubmissionStatistic>();
         private const int MaximumLineLength = 100;
         private readonly TcpListener ProductionListener = new TcpListener(IPAddress.IPv6Any, 1337);
         private readonly TcpListener DebugListener = new TcpListener(IPAddress.IPv6Any, 1338);
         private readonly ILogger Logger;
         private readonly JsonConfiguration Configuration;
         private readonly IServiceProvider ServiceProvider;
-        private readonly EnoStatistics Statistics;
+        private readonly EnoStatistics EnoStatistics;

```

Comment:
Naming the variable exactly the same as the type is not bad in c#?

Reply:
Nope, the compiler knows that the type comes before the variable name. Problems arise if types and namespaces share names, because you can specify types with their namespace